src/send_future.rs

```diff
@@ -62,21 +54,53 @@ impl<R, F> SendFuture<R, F, ResolveToHandlerReturn> {
     }
 }
 
-impl<R> SendFuture<R, BoxFuture<'static, Receiver<R>>, ResolveToHandlerReturn> {
-    pub(crate) fn sending_boxed<F>(send_fut: F) -> Self
+pub(crate) mod private {
+    pub trait SetPriority {
+        fn set_priority(&mut self, priority: u32);
+    }
+}
+
+impl<R, F, TResolveMarker> SendFuture<R, F, TResolveMarker>
+where
+    F: SetPriority,
```

Comment:
I think I would have probably created two impl blocks instead of a trait but that is more of a style question.

This solution might be slightly more expensive to compile.

Reply:
I originally planned to use SetPriority to let BroadcastFuture with erased actor type set priority too. I can push that branch but it didn't work out for other reasons that I discussed, so we can move to two impl blocks maybe

Reply:
I think we can leave this as-is for now since when messagechannels get refactored they could use SetPriority too